db/seedRavenDB.js

```diff
@@ -0,0 +1,205 @@
+    process.stdout.write(`${parseFloat(i / 1e7 * 100).toFixed(2)}% of descriptions seeded.`);
+  }
+
+
+  return true;
+}
+
+/**
+ * Seeds database with all games_genres CSV files from csvDir
+ * @param {String} csvDir
+ * @param {Array} filenames: array of filename strings matching /games_genres_\d+\.csv/ at csvDir
+ * @returns {Promise->Boolean} true if seed operation successful
+ */
+const seedGameGenres = async () => {
+  // Generate data and bulkInsert in chunks of roughly 10k (1000 * range(3,5) ~= 10k)
+  const GENERATE_LIMIT = 1000;
+  let startId = 1;
+
+  for (let i = 0; i < 1e7; i += GENERATE_LIMIT) {
+    let documents = generateGamesGenresWithoutWrite(GENERATE_LIMIT, 53, startId);
+
+    const bulkInsert = store.bulkInsert();
+
+    for (let j = 0; j < documents.length; j++) {
+      await bulkInsert.store(documents[j]);
+    }
+
+    startId += documents.length;
+    await bulkInsert.finish();
+
+    process.stdout.clearLine();
+    process.stdout.cursorTo(0);
+    process.stdout.write(`${parseFloat(i / 1e7 * 100).toFixed(2)}% of games_genres seeded. Current id: ${startId}`);
+  }
+
+  return true;
+}
+// const seedGameGenres = async (csvDir, filenames) => {
+//   for (let i = 0; i < filenames.length; i++) {
+//     // Acquire parsed genre.csv records from input csvDir
+//     let records = await parseCSVFile(`${csvDir}/${filenames[i]}`);
+
+//     // Insert records from parsed CSV as bulkInsert operation
+//     const bulkInsert = store.bulkInsert();
+//     for (const ggObj of records) {
+//       const gg = new GameGenre(ggObj.id, ggObj.product_id, ggObj.genre_id);
+//       await bulkInsert.store(gg);
+//     }
+
+//     await bulkInsert.finish();
+
+//     process.stdout.clearLine();
+//     process.stdout.cursorTo(0);
+//     process.stdout.write(`${i} / ${filenames.length} games_genres files seeded`);
+//   }
+
+//   return true;
+// }
+
+/**
+ * Main seeding function, called on running node db/seedRavenDB.js.
+ * Parses CSV from input directory, then inserts documents into RavenDB.
+ * NOTES:
+ * - databases 'item_descriptions' and 'item_descriptions_test' must be created beforehand via studio GUI at http://localhost:${RDBPORT} (default 8080)
+ * - collections are parsed automatically through inserting items as class instances
+ * - id is specified in all insertion calls to set deterministic primary keys, instead of allowing RavenDB uuid generation
+ */
+const seed = async () => {
+  let defaultCSVDirPath = process.env.NODE_ENV === 'test' ?
+    path.resolve(__dirname, '..', 'tests', 'fixtures') :
+    path.resolve(__dirname, 'data-gen', 'csv');
+
+  let args = JSON.parse(process.env.npm_config_argv).original;
+  let csvDir = args.find(elt => /--dir=/.test(elt));
+
+  // Check if valid csvDir (ensuring path originates from project root instead of db/ dir)
+  if (csvDir) {
+    csvDir = path.resolve(__dirname, '..', csvDir.replace('--dir=', ''));
+    try {
+      await fs.promises.stat(csvDir);
```

Comment:
Christina, I am curious if it it's more a style to implementation when doing fs.promises.stat vs const fs = require('fs').promises in the beginning and using fs.stat later? I haven't seen many examples with the latter as I know the native api fs promises module do not have all of the fs methods.

Reply:
The first case would be if you had to use non-promise modules as well, but I don't think I use those in this file. It's a matter of preference 👍